extension/lib/shopify.api.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @param {object} config
+ * @return {{}}
+ */
+module.exports = function (config) {
+  const module = {}
+  const shopDomain = 'https://' + config.shopifyShopAlias + '.myshopify.com'
+
+  module.getCollectionListingUrl = () => {
+    return shopDomain + '/admin/collection_listings.json'
+  }
+
+  module.getCollectionProductCountUrl = (collectionId) => {
+    return shopDomain + '/admin/products/count.json?collection_id=' + collectionId
+  }
+
+  module.getGraphQlUrl = () => {
+    return shopDomain + '/api/graphql'
+  }
+
+  /**
+   * Adds necessary Header-Informations which are needed for the Request
+   * @returns {object}
+   */
+  module.getAdminApiRequestHeader = () => {
+    return {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Shopify-Access-Token': config.shopifyAccessToken
+      }
+    }
+  }
+
+  /**
+   * Adds necessary Header-Informations which are needed for the Request
+   * @param body
+   * @returns {object}
+   */
+  module.getGraphQlApiRequestHeader = (body) => {
+    return {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Shopify-Storefront-Access-Token': config.shopifyStorefrontAccessToken
```

Comment:
The storefront access token is not available in the config and should not be placed there. Instead it needs to be requested, using the Shopify Admin API and placed in the application storage.

Reply:
Done